test/arbcall-test.jl

```diff
@@ -0,0 +1,130 @@
+                                              ("const mag_t x", "x", true, Arblib.Mag, Ref{Arblib.Mag}),
+                                              ("const arf_t x", "x", true, Arf, Ref{Arf}),
+                                              ("const arb_t x", "x", true, Arb, Ref{Arb}),
+                                              ("const acb_t x", "x", true, Acb, Ref{Acb}),
+                                    )
+        arg = Arblib.Carg(str)
+        @test Arblib.name(arg) == name
+        @test Arblib.isconst(arg) == isconst
+        @test Arblib.jltype(arg) == type
+        @test Arblib.ctype(arg) == ctype
+    end
+
+    # Unsupported types
+    for str in ("fmpz_t x",
+                "fmpq_t x",
+                "mag_ptr res",
+                "arb_ptr res",
+                "acb_ptr res",
+                "const fmpz_t x",
+                "const fmpq_t x",
+                "mag_srcptr res",
+                "arb_srcptr res",
+                "acb_srcptr res",
+
+                # Internal types
+                "mpfr_rnd_t rnd",
+                "mp_limb_t lo",
+                "mp_bitcnt_t r",
+                "mp_ptr ycos",
+                "mp_srcptr x",
+                )
+        @test_throws KeyError arg = Arblib.Carg(str)
+    end
+
+    # Parsed incorrectly
+    for str in ("const char * inp",
+                )
+        @test_throws KeyError arg = Arblib.Carg(str)
+    end
+
+
+    # Parse errors
+    for str in (# Internal types
+                "mp_limb_t * error",
+                "mp_bitcnt_t * Qexp",
+                )
+        @test_throws ArgumentError arg = Arblib.Carg(str)
+    end
+end
+
+@testset "jlfname" begin
+    for (arbfname, name) in (# Supported types
+                             ("mag_set", :set),
+                             ("mag_set_d", :set),
+                             ("mag_set_ui", :set),
+                             ("arf_set", :set),
+                             ("arf_set_ui", :set),
+                             ("arf_set_si", :set),
+                             ("arf_set_d", :set),
+                             ("arb_set", :set),
+                             ("arb_set_arf", :set),
+                             ("arb_set_si", :set),
+                             ("arb_set_ui", :set),
+                             ("arb_set_d", :set),
+                             ("acb_set", :set),
+                             ("acb_set_ui", :set),
+                             ("acb_set_si", :set),
+                             ("acb_set_d", :set),
+                             ("acb_set_arb", :set),
+                             ("acb_set_si_si", :set),
+                             ("acb_set_d_d", :set),
+                             ("acb_set_arb_arb", :set),
+
+                             # Unsupported types
+                             ("arf_set_fmpz", :set_fmpz),
+                             ("arf_set_mpfr", :set_mpfr),
```

Comment:
I think we should be able to support MPFR since these are just `BigFloat`s.

Reply:
Yes, this should be a simple change. I also think it could be nice to support `fmpz` by creating a very simple type for it which only purpose is to convert between `BigInt` and `fmpz`, but I think it is better to that leave for later.

Reply:
I think some hand-written `set!` already use BigFloat:
https://github.com/kalmarek/Arblib.jl/blob/981f097003d484d86be2f6f442e105e4818eb673/src/constructors.jl#L7
so let's just add it to the "supported" list